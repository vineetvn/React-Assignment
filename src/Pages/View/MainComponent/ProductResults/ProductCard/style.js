import styled from 'styled-components';
import {HiShoppingCart} from 'react-icons/hi'

export const Div = styled.div `
    margin: 15px;
    display: flex;
    flex-direction: column;
    height: 320px;
    width: 250px;
    border: 1px solid ${props => props.theme.color.silverGrey};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    box-sizing: border-box;
    text-align: center;
    position: relative;

    @media only screen and (max-width: 1000px) {
        height: 320px;
        width: 180px;
    } 

    @media only screen and (max-width: 685px) {
        height: 300px;
        width: 70%;
        margin: 15px auto;
    } 

    @media only screen and (max-width: 450px) {
        height: 300px;
        width: 90%;
        margin: 15px auto;
    }
`

export const ShoppingCart = styled(HiShoppingCart) `
    color: white;
    background-color: black;
    padding: 10px;
    position: absolute;
    right: 0;
    top: 190px;
`
export const Img = styled.img `
    height: 225px;
    width: 100%;
    display: block;
`

export const ImageDetails = styled.div `
    height: 50px;
    box-sizing: border-box;
    font-family: ${props => props.theme.fontFamily.Montserrat}
`

export const H4 = styled.h4 `
    margin: 15px;
    text-align: center;
    box-sizing: border-box;
`

export const Price = styled.span `
    margin: 15px;
    text-align: center;
    box-sizing: border-box;
    margin-left: 5px;
`

export const OfferPrice = styled.span `
    margin: 15px;
    text-decoration: line-through;
    text-align: center;
    box-sizing: border-box;
`