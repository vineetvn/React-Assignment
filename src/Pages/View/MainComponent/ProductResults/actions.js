export const setId = (data) => {
    return {
        type: 'SET_ID',
        data: data
    }
}