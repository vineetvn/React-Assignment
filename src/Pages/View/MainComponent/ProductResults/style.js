import styled from 'styled-components';

export const OuterDiv = styled.div `
    flex: 70%;
    width: 100%;
    height: auto;
    display: flex;

    @media only screen and (max-width: 500px) {
        flex: 100%;
    }
`

export const MainSection = styled.div `
    flex: 75%;
`

export const Range = styled.div `
    margin: 15px 0;
    width: 100%;
    font-family: ${props => props.theme.fontFamily.Montserrat};
    display: flex;
    @media only screen and (max-width: 1000px) {
        flex-direction: column;
    }
`

export const P = styled.p `
    margin: 50px;
    font-family: ${props => props.theme.fontFamily.Montserrat};
`

export const Filter = styled.div `
    margin: 0 20px;
    @media only screen and (max-width: 1000px) {
        margin: 25px 0;
    }

`

export const Div = styled.div `
    height: auto;
    display: flex;
    flex-wrap: wrap;
    margin: 20px;

    @media only screen and (max-width: 500px) {
        flex-wrap: nowrap;
        flex-direction: column;
    }
`

export const Div1 = styled.div `
    width: 100%;
`

export const Span = styled.span `
    margin: 0 20px 0 40px;

    @media only screen and (max-width: 700px) {
        margin: 0 40px;
    }
    @media only screen and (max-width: 350px) {
        display: none;
    }
`

export const Input = styled.input `
    -webkit-appearance: none;
    background: grey; 
    outline: none;
    height: 5px;
    margin: 0 20px;
    &::-webkit-slider-thumb {
        -webkit-appearance: none;
        width: 15px;
        height: 15px;
        background: black;
        cursor: pointer;
    }

    @media only screen and (max-width: 700px) {
        margin: 0 15px;
    }

    @media only screen and (max-width: 350px) {
        margin: 0 30px;
    }

`

export const Search = styled.input `
    border: 1px solid black;
    height: 20px;
    font-family: ${props => props.theme.fontFamily.Lato};

    @media only screen and (max-width: 700px) {
        margin: 0 0 0 15px;
    }

    @media only screen and (max-width: 350px) {
        margin: 0 30px;
    }

`