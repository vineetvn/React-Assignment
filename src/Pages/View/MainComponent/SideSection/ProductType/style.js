import styled from 'styled-components';
import {Link} from 'react-router-dom';

export const Div = styled.div `
    flex:25%auto;
    margin: 130px 50px 0 50px;
    display: flex;
    flex-direction: column;
    text-align: center;
    background-color: ${props => props.theme.color.silverGrey};
    color: ${props => props.theme.color.darkCharcoal};
    font-size: ${props => props.theme.fontSize.productName};
    font-family: ${props => props.theme.fontFamily.Montserrat};
    padding: 30px 10px 10px 10px;

    @media only screen and (max-width: 1000px) {
        margin: 25px;
        flex: 25%;

    }
`

export const StyledLink = styled(Link) `
    text-decoration: none;
    color: ${props => props.theme.color.red};
    font-weight: ${props => props.theme.fontWeight.bold};
    padding-bottom: 20px;
`