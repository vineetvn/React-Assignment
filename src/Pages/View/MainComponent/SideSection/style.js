import styled from 'styled-components';

import Woman from '../../../../Assets/Images/Woman.jpg';

export const ImageWrapper = styled.div `
    background: url('${Woman}');
    flex: 75%;
    background-repeat: no-repeat;
    background-position: center;
    height: 100%auto;
    margin: 50px;

    @media only screen and (max-width: 1000px) {
        margin: 25px;
        flex: 75%;
        height: 100%;
    }    
`

export const Div = styled.div `
    display: flex;
    flex-direction: column;
    flex: 30%;

    @media only screen and (max-width: 1000px) {
        margin-top: 100px;
    } 
`