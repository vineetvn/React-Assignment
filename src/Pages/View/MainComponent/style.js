import styled from 'styled-components';

export const Div = styled.div `
    display: flex;
    height: auto;
    width: 100%;

    @media only screen and (max-width: 600px) {
        flex-direction: column;
    }
`