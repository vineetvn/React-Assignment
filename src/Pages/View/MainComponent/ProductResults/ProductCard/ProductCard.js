import React from 'react'

import {Div, ImageDetails, H4, Price, Img, OfferPrice, ShoppingCart} from './style'

function ProductCard(props) {
    return (
        <Div onClick={props.clicked}>
            <Img src= {require(`../../../../../Assets/Images/${props.image}`)}></Img>
            <ShoppingCart />
            <ImageDetails>
            <H4>{props.name}</H4>
            {props.discount ? <OfferPrice>${props.discount}</OfferPrice>: null}<Price>${props.price}</Price>
            </ImageDetails>
        </Div>
    )
}

export default ProductCard
