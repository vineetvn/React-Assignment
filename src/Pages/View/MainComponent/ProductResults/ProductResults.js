import React, { useEffect, useState } from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router-dom';

import {Div, OuterDiv, Range, Span, P, Input, Search, Filter, Div1} from './style';
import ProductCard from './ProductCard/ProductCard';
import {setId} from './actions'

export function ProductResults(props) {

    const [value, setValue] = useState(200);
    const [filteredProducts, setFilteredproducts] = useState(null);
    const [filtered , setFiltered] = useState(null);
    const [searchValue, setSearchValue] = useState('');

    useEffect(() => {
        let filter = props.relatedProducts.filter(prod => {
            if(prod.price <= value) {
                return true;
            }else return false;
        })
        setFiltered(filter)
    }, [props.relatedProducts, value])

    useEffect(() => {
        let searchFilter = props.relatedProducts.filter(prod => {
            let name = prod.name.toLowerCase();
            name = name.trim();
            if(name.includes(searchValue)) {
                return true;
            }else return false;
        })
        setFiltered(searchFilter)
    }, [props.relatedProducts, searchValue])

    const sliderChangeHandler = (e) => {
        setValue(e.target.value)
        setFilteredproducts(filtered)
    }

    const changeHandler = (e) => {
        e.preventDefault();
        setSearchValue(e.target.value.toLowerCase().trim())
        if(e.target.value) {
            setFilteredproducts(filtered)
        }else setFilteredproducts(props.relatedProducts)

    }

    let products;
    if(filteredProducts) {
        products = filteredProducts.map((prod) => {
            return <ProductCard key= {prod.id} image={prod.url} name={prod.name} price= {prod.price} />
        })
    }else products = props.relatedProducts.map((prod) => {
        return <ProductCard key= {prod.id} image={prod.url} name={prod.name} price= {prod.price} discount = {prod.offerprice} />
    })

    return (
        <OuterDiv>
            <Div1> 
                <Div1>
                <P>Showing 1-{products.length} of 23 results</P>
                {!(props.locationState + 1) ? <Range>
                                                <Filter>
                                                    <Span>Filter :</Span>
                                                    <Input type="range" min="1" max="200" 
                                                    value= {`${value}`} 
                                                    onChange= {(e) =>sliderChangeHandler(e)} />
                                                    <span>$0 to ${value}</span>
                                                </Filter>
                                                <div>
                                                    <Span>Search :</Span>
                                                    <Search type='input' 
                                                    placeholder='Search your Products'
                                                    value={`${searchValue}`}
                                                    onChange={(e) => changeHandler(e)} />
                                                </div>
                                            </Range> : null}
                </Div1>
                <Div>
                    {products}
                </Div>
            </Div1>
        </OuterDiv>
        )
}

const mapStateToProps = state => {
    return {
        relatedProducts: state.view.relatedProducts,
        locationState: state.view.locationState
    }
}

const mapDispatchToProps = dispatch => {
    return {
        setId: (data) => dispatch(setId(data))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(ProductResults))
