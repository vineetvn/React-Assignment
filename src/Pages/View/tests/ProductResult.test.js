import Adapter from 'enzyme-adapter-react-16';
import {configure} from 'enzyme';
import { configureMockStore, mountRenderWithRouter, shallowRender } from '../../../testUtils';
import {ProductResults as ProdResult} from '../MainComponent/ProductResults/ProductResults';
import ProductResults from '../MainComponent/ProductResults/ProductResults'
import { Filter, P, Range, Search } from '../MainComponent/ProductResults/style';
import ProductCard from '../MainComponent/ProductResults/ProductCard/ProductCard';

configure({adapter: new Adapter()});

describe('ProductResult', () => {
    let props = {
        relatedProducts: [
            {
                id: 0,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
            {
                id: 1,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
            {
                id: 2,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
            {
                id: 3,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
        ],
        locationState: 0
    };
    let wrapper;

    beforeEach(() => {
        wrapper = shallowRender(ProdResult, false, props)
    })

    it('should render the component', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('should render the filter component when locationstate is null', () => {
        props.locationState = null;
        expect(wrapper.find(Filter))
    });

    it('should render the search component when locationstate is null', () => {
        props.locationState = null;
        expect(wrapper.find(Search))
    });

    it('should not render the search and filter components when related products are shown', () => {
        expect(wrapper.find(Range)).toHaveLength(0)
    });

    it('should render the paragraph showing number of results dynamically', () => {
        expect(wrapper.find(P))
    });

    it('should render 4 Product cards', () => {
        expect(wrapper.find(ProductCard)).toHaveLength(4)
    });
});

describe('Product Result should render deeply', () => {
    let initialState = {
        view: {
            relatedProducts: [
                {
                    id: 0,
                    url: 'bagpack.jpg',
                    name: 'backpack',
                    price: '$10'
                },
                {
                    id: 1,
                    url: 'bagpack.jpg',
                    name: 'backpack',
                    price: '$10'
                },
                {
                    id: 2,
                    url: 'bagpack.jpg',
                    name: 'backpack',
                    price: '$10'
                },
                {
                    id: 3,
                    url: 'bagpack.jpg',
                    name: 'backpack',
                    price: '$10'
                },
            ],
            locationState: 0
        }
    }

    let props = {
        relatedProducts: [
            {
                id: 0,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
            {
                id: 1,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
            {
                id: 2,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
            {
                id: 3,
                url: 'bagpack.jpg',
                name: 'backpack',
                price: '$10'
            },
        ],
        locationState: 0,
        clicked: false,
        image: 'backpack.jpg',
        name: 'backpack',
        discount: 0,
        price: '$10'
    };

    let store = configureMockStore(initialState);
    let wrapper;

    beforeEach(() => {
        wrapper = mountRenderWithRouter(ProductResults, store, props)
    })

    it('should render deeply', () => {
        expect(wrapper).toMatchSnapshot();
    });
});