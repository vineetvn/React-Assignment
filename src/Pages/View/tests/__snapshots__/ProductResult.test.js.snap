// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Result should render deeply should render deeply 1`] = `undefined`;

exports[`ProductResult should render the component 1`] = `ShallowWrapper {}`;
